### PR TITLE
A bunch of (bug)fixes

### LINKED ISSUE #478: Harmonization/regex for malware.name error on "yahoo!"
```shell
  File "/home/intelmq/intelmq-src/intelmq/bots/parsers/cleanmx/parser_phishing.py", line 73, in process
    event.add(key, value)
  File "/home/intelmq/intelmq-src/intelmq/lib/message.py", line 125, in add
    raise exceptions.InvalidValue(key, value, reason=valid_value[1])
intelmq.lib.exceptions.InvalidValue: invalid value 'yahoo!' (<class 'str'>) for key 'malware.name' regex did not match.
```

regex change? what do you propose?

### PR DESCRIPTION
**Pull request by sebix**
@dmth @swilde Please review the shadowserver part. (except `recover_line`, it will be rewritten by #618)

Fixes #478 #576 #587 #590 #594 #597 #599 #600 #603 #607 (x2)

### CODE REVIEW

File: intelmq/bots/parsers/shadowserver/parser.py [hunk 1]
```diff
@@ -134,6 +136,8 @@ def parse_line(self, row, report):
         # extra if an add operation failed
         for item in conf.get('optional_fields'):
             intelmqkey, shadowkey = item[:2]
+            if shadowkey not in fields:  # key does not exist in data (not even in the header)
```
Comment (swilde) [inline intelmq/bots/parsers/shadowserver/parser.py]: I wonder if it wouldn't be appropriate to write at least a warning to
the log file, as this situation (an key expected by the parser
configuratoin is not presend in the feed data) is something an admin
(or developer) might want to know about...

Comment (sebix) [inline intelmq/bots/parsers/shadowserver/parser.py]: True

**Comment by dmth** [comment]
In which order do you plan to apply pull/620 and pull/618? (I think it's not important, there will be a merge conflict in `recover_line` anyway)

Are you aware that pull/618 will restore the (from our point) correct naming of keys in shadowserver parser config?

**Comment by aaronkaplan** [comment]
> On 27 Jul 2016, at 17:06, Dustin Demuth <notifications@github.com> wrote:
> 
> In which order do you plan to apply pull/620 and pull/618? (I think it's not important, there will be a merge conflict in recover_line anyway)
> 
> Are you aware that pull/618 will restore the (from our point) correct naming of keys in shadowserver parser config?
> 

Yes, sorry but then we won't pull that in.

**Comment by sebix** [comment]
> In which order do you plan to apply pull/620 and pull/618? (I think it's not important, there will be a merge conflict in recover_line anyway)

Don't know. But doesn't matter I think.

> Are you aware that pull/618 will restore the (from our point) correct naming of keys in shadowserver parser config?

I will remove this commit before merging. Once the tests in #618 are fixed, I merge both PRs (manually).

@aaronkaplan can you please fix the keys in the docs and document how the keys are named?

File: intelmq/bots/parsers/shadowserver/parser.py (hunk 1 above)
Comment (sebix) [inline intelmq/bots/parsers/shadowserver/parser.py]: Hm, but as it's optional it's kind of expected. Would prefer info instead. What do you think?

**Comment by aaronkaplan** [comment]
> On 27 Jul 2016, at 17:13, Sebastian <notifications@github.com> wrote:
> 
> In which order do you plan to apply pull/620 and pull/618? (I think it's not important, there will be a merge conflict in recover_line anyway)
> 
> Don't know. But doesn't matter I think.
> 
> Are you aware that pull/618 will restore the (from our point) correct naming of keys in shadowserver parser config?
> 
> I will remove this commit before merging. Once the tests in #618 are fixed, I merge both PRs (manually).
> 
> @aaronkaplan can you please fix the keys in the docs and document how the keys are named?

Will do.

**Pull request merged**